src/components/uri/Visualization.tsx

```diff
@@ -246,24 +254,31 @@ export class Visualization extends React.Component<IVisualizationProps, IVisuali
                 return this.props.fetchVisObject(visualizationUri);
             })
             .then((mdObject: VisualizationObject.IVisualizationObject) => {
-                const translatedPopSuffix = ' - previous year'; // TODO change hardcoded PoP suffix with new visObj
-
-                const visObjTotals = mdObject.content.buckets.totals;
-                const totals = !isEmpty(visObjTotals) ?
-                    visObjTotals.map(visObjTotal => visObjTotal.total) : [];
-
-                const { afm, resultSpec } = toAfmResultSpec(mdObject.content, translatedPopSuffix);
-                const dateFilter = getDateFilter(filters);
-                const attributeFilters = getAttributeFilters(filters);
-                const afmWithFilters = AfmUtils.appendFilters(afm, attributeFilters, dateFilter);
-                return this.adapter.createDataSource(afmWithFilters)
-                    .then((dataSource: IDataSource) => {
-                        return {
-                            type: mdObject.content.type,
-                            dataSource,
-                            resultSpec,
-                            totals
-                        };
+                const visualizationClass: string = get(mdObject, 'content.visualizationClass.uri', '');
+                return this.props.fetchVisualizationClass(visualizationClass).then((visualizationClass) => {
+                    const { afm, resultSpec } = toAfmResultSpec(mdObject.content);
+
+                    const mdObjectTotals = MdObjectHelper.getTotals(mdObject);
+
+                    const dateFilter = getDateFilter(filters);
+                    const attributeFilters = getAttributeFilters(filters);
+                    const afmWithFilters = AfmUtils.appendFilters(afm, attributeFilters, dateFilter);
+
+                    const visualizationType: VisType = getVisualizationType(get(visualizationClass, 'content.url', ''));
```

Comment:
I think it would be better to put 
```
get(visualizationClass, 'content.url', '')
```
into `getVisualizationType` method and hide `lodash/get` usage

Reply:
moved to helper